src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -232,15 +232,41 @@ Namespace Microsoft.CodeAnalysis.Semantics
         End Function
 
         Private Function GetVariableDeclarationStatementVariables(declarations As ImmutableArray(Of BoundLocalDeclarationBase)) As ImmutableArray(Of IVariableDeclaration)
+            ' Group the declarations by their VariableDeclaratorSyntaxes. The issue we're compensating for here is that the
+            ' the declarations that are BoundLocalDeclaration nodes have a ModifiedIdentifierSyntax as their syntax nodes,
+            ' not a VariableDeclaratorSyntax. We want to group BoundLocalDeclarations by their parent VariableDeclaratorSyntax
+            ' nodes, and deduplicate based on that. As an example:
+            '
+            ' Dim x, y = 1
+            '
+            ' This is an error scenario, but if we just use the BoundLocalDeclaration.Syntax.Parent directly, without deduplicating,
+            ' we'll end up with two IVariableDeclarations that have the same syntax node. So, we group by VariableDeclaratorSyntax
+            ' to put x and y in the same IVariableDeclaration
+
+            Dim groupedDeclarations = declarations.GroupBy(Function(decl)
+                                                               If decl.Kind = BoundKind.LocalDeclaration Then
+                                                                   Debug.Assert(decl.Syntax.Parent.IsKind(SyntaxKind.VariableDeclarator))
+                                                                   Return decl.Syntax.Parent
+                                                               Else
+                                                                   Return decl.Syntax
+                                                               End If
+                                                           End Function)
+
             Dim builder = ArrayBuilder(Of IVariableDeclaration).GetInstance()
-            For Each base In declarations
-                If base.Kind = BoundKind.LocalDeclaration Then
-                    Dim declaration = DirectCast(base, BoundLocalDeclaration)
-                    builder.Add(OperationFactory.CreateVariableDeclaration(declaration.LocalSymbol, Create(declaration.InitializerOpt), _semanticModel, declaration.Syntax))
-                ElseIf base.Kind = BoundKind.AsNewLocalDeclarations Then
-                    Dim asNewDeclarations = DirectCast(base, BoundAsNewLocalDeclarations)
+            For Each declGroup In groupedDeclarations
+                If declGroup.First().Kind = BoundKind.AsNewLocalDeclarations Then
```

Comment:
dim first = declGroup.First() and reuse rather than running enumerable multiple times?

Reply:
Done.